File: c++/Node.cpp
#include <iostream>
#include <string>
#include "Node.h"

class Node{
public:
    Node(std::string i_exept){
        setlocale(LC_ALL,"RUS");
        exept = i_exept;
    }

    void add_child(Node &left, Node &right){
        leftChild = &left;
        rightChild = &right;
    }

    void show_exeption(){
        if (leftChild != NULL && rightChild != NULL)
        {
            std::cout<<exept<<" (enter Y/N)"<<std::endl;
        }
        else 
        {
            std::cout<<exept;
        }
    }

    Node get_left_child(){
        return *leftChild;
    }
    Node get_right_child(){
        return *rightChild;
    }

    bool have_child(){
        return leftChild != NULL && rightChild != NULL;
    }

private:
    std::string exept;
    Node* leftChild = NULL;
    Node* rightChild = NULL;
};

File: python/node.py
class Node:
    def __init__(self, exeption:str):
        self.exept = exeption
        self.leftChild = None
        self.rightChild = None
        pass
    def __call__(self, leftChild, rightChild):
        self.leftChild = leftChild
        self.rightChild = rightChild
        pass
    def show_exeption(self):
        if self.leftChild != None and self.rightChild != None:
            print(self.exept + ' (введите Y/n)')
        else:
            print(self.exept)
        pass
    def get_left_child(self):
        return self.leftChild
    def get_right_child(self):
        return self.rightChild
    def node_have_childs(self):
        return self.leftChild != None and self.rightChild != None